src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -1009,4 +1010,61 @@ public void setSecurityAnalysisServerBaseUri(String securityAnalysisServerBaseUr
     public void setActionsServerBaseUri(String actionsServerBaseUri) {
         this.actionsServerBaseUri = actionsServerBaseUri;
     }
+
+    public Mono<Void> stopSecurityAnalysis(String studyName, String userId) {
+        Objects.requireNonNull(studyName);
+        Objects.requireNonNull(userId);
+
+        return studyRepository.findStudy(userId, studyName).flatMap(entity -> {
+            UUID resultUuid = entity.getSecurityAnalysisResultUuid();
+
+            String receiver;
+            try {
+                receiver = URLEncoder.encode(objectMapper.writeValueAsString(new Receiver(studyName, userId)), StandardCharsets.UTF_8);
```

Comment:
don't encode manually, let UriComponentsBuilders encode. Also, is it safe to pass encoded urls to webclient ? (I'm asking because it was not with RestTemplate... maybe webclient has a better api?)